lib/board.ex

```diff
@@ -18,6 +18,11 @@ defmodule Board do
     Enum.any?([get_all_rows_status(board), get_all_columns_status(board), get_both_diagonals_status(board)])
   end
 
+  @spec has_game_tied?(map) :: boolean
+  def has_game_tied?(board) do
+    !Enum.member?(Map.values(board), :empty)
+  end
```

Comment:
Something to be cautious of – this implementation depends on a prior call to `Board.has_player_won?(board)` to be correct in some cases. See [temporal coupling](https://blog.ploeh.dk/2011/05/24/DesignSmellTemporalCoupling/).

Suppose the following board exists where X plays the last (winning) move:

```
X | X | X
---------
O | O | X
---------
X | O | O
```

If this board is provided as an argument to `has_game_tied?/1` what is the returned value?

Reply:
Another way to think about this. Are all boards will no empty spaces tied boards?